src/test/lib/HandshakeTest.cpp

```diff
@@ -4033,3 +4033,202 @@ QuicTestHandshakeSpecificLossPatterns(
         Listener.LastConnection->Shutdown(0, QUIC_CONNECTION_SHUTDOWN_FLAG_SILENT);
     }
 }
+
+struct ServerMultiAcceptContext {
+    uint32_t StartedConnectionCount;
+    uint32_t MaxConnections;
+    CxPlatEvent StartedEvent;
+    TestConnection** Connections;
+
+    _Function_class_(NEW_CONNECTION_CALLBACK)
+    static
+    bool
+    ListenerMultiAcceptConnection(
+        _In_ TestListener* Listener,
+        _In_ HQUIC ConnectionHandle
+        )
+    {
+        auto* This = (ServerMultiAcceptContext*)Listener->Context;
+        if (This->StartedConnectionCount > This->MaxConnections) {
+            return false;
+        }
+        This->Connections[This->StartedConnectionCount++] =
```

Comment:
What happens if/when this is called in parallel on two threads/cpus?

Reply:
That's when the fun begins.
(good catch, I'll use an interlocked increment)